@xen-orchestra/web/src/types/xo/new-vm.type.ts

```diff
@@ -13,6 +13,14 @@ export interface Vdi {
   userdevice?: string
 }
 
+export interface Vif {
+  tempId?: string
+  interface?: XoNetwork['id']
+  macAddress?: string
+  device?: string
+  destroy?: boolean
+}
+
 export interface NetworkInterface {
```

Comment:
```suggestion
export interface NetworkInterface {
  interface?: XoNetwork['id']
  macAddress?: string
  device?: string
  destroy?: boolean
  }
```

Since interface `Vif` is the same as `NetworkInterface` you can keep `NetworkInterface` and add optional fields